homeassistant/components/google_assistant/smart_home.py

```diff
@@ -79,18 +79,17 @@
         EVENT_SYNC_RECEIVED, {"request_id": data.request_id}, context=data.context
     )
 
+    agent_user_id = data.config.agent_user_id or data.context.user_id
```

Comment:
Why not always take it from context? That way we can remove `config.agent_user_id` ?

Reply:
And we just add a check that blows up if no user_id in the context when processing messages.